### PR TITLE
Ocrvs 6553 data seeder fix

### LINKED ISSUE #6553: Data seeding fails if done too quickly
**Describe the bug**

**Which feature of OpenCRVS your bug concern?**
Data seeding in development environment

**To Reproduce**
Steps to reproduce the behaviour:

1. Clear all data from your environment
2. Quickly run `yarn seed:dev` right after

**Expected behaviour**
Environment should seed successfully. 

**Actual behaviour**
```
@opencrvs/data-seeder: $ ts-node -r tsconfig-paths/register src/index.ts
@opencrvs/data-seeder: Seeding roles
@opencrvs/data-seeder: Seeding locations
@opencrvs/data-seeder: Seeding users
@opencrvs/data-seeder: [
@opencrvs/data-seeder:   {
@opencrvs/data-seeder:     message: "Something went wrong on user-mgnt service. Couldn't create user",
@opencrvs/data-seeder:     locations: [ [Object] ],
@opencrvs/data-seeder:     path: [ 'createOrUpdateUser' ],
@opencrvs/data-seeder:     extensions: { code: 'INTERNAL_SERVER_ERROR', exception: [Object] }
@opencrvs/data-seeder:   }
@opencrvs/data-seeder: ]
@opencrvs/data-seeder: error Command failed with exit code 1.
```

There's another bug here as well: if you run the command again, you'll see the script fail already in 

```
@opencrvs/data-seeder: Seeding locations
@opencrvs/data-seeder: {
@opencrvs/data-seeder:   statusCode: 500,
@opencrvs/data-seeder:   error: 'Internal Server Error',
@opencrvs/data-seeder:   message: 'An internal server error occurred'
@opencrvs/data-seeder: }
@opencrvs/data-seeder: error Command failed with exit code 1.
@opencrvs/data-seeder: info Visit https://yarnpkg.com/en/docs/cli/run for documentation abo
```

This is because the seed script now tries to send an empty array of locations for seeding. 

**OpenCRVS Core Version:**
1.4.0, 1.3.2 (Git branch: master)

**Country Configuration Version:**
1.4.0, 1.3.2 (Git branch: master)

## Tech tasks
- Find a better way to verify OpenHIM is ready to take in requests. There must be an API endpoint that can be used for this.

### PR DESCRIPTION
**Pull request by tahmidrahman-dsi**

### CODE REVIEW

File: packages/data-seeder/src/users.ts [hunk 1]
```diff
@@ -124,68 +126,80 @@ async function getOfficeIdFromIdentifier(identifier: string) {
...
+      role,
+      primaryOfficeId: officeIdentifier,
+      username,
+      ...user
+    } = userMetadata
+    if (await userAlreadyExists(token, username)) {
+      console.log(
+        `User with the username "${username}" already exists. Skipping user "${username}"`
+      )
+      return
```
Comment (Zangetsu101) [inline packages/data-seeder/src/users.ts:163]:
```suggestion
      continue
```

Comment (Zangetsu101) [reply packages/data-seeder/src/users.ts:163]: Same for the 2 return statements below

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/data-seeder/src/users.ts [hunk 2]
```diff
@@ -124,68 +126,80 @@ async function getOfficeIdFromIdentifier(identifier: string) {
...
-            user: userPayload
-          }
-        })
-      })
-      parseGQLResponse(await res.json())
-    })
-  )
+      res = await callCreateUserMutation(token, userPayload)
+      jsonRes = await res.json()
+    } while (tryNumber < MAX_RETRY && 'errors' in jsonRes)
```
Comment (Zangetsu101) [inline packages/data-seeder/src/users.ts:202]: Let's add some delay between each try

**Comment by rikukissa** [comment]
I can still reproduce the issue :/ 

```
yarn run v1.22.4
$ ACTIVATE_USERS=true lerna run seed --stream --scope @opencrvs/data-seeder
lerna notice cli v5.6.2
lerna notice filter including "@opencrvs/data-seeder"
lerna info filter [ '@opencrvs/data-seeder' ]
lerna info Executing command in 1 package: "yarn run seed"
@opencrvs/data-seeder: $ ts-node -r tsconfig-paths/register src/index.ts
@opencrvs/data-seeder: Seeding roles
@opencrvs/data-seeder: Seeding locations
@opencrvs/data-seeder: Seeding users
@opencrvs/data-seeder: Trying again for time:  2
@opencrvs/data-seeder: Trying again for time:  3
@opencrvs/data-seeder: [
@opencrvs/data-seeder:   {
@opencrvs/data-seeder:     message: "Something went wrong on user-mgnt service. Couldn't create user",
@opencrvs/data-seeder:     locations: [ [Object] ],
@opencrvs/data-seeder:     path: [ 'createOrUpdateUser' ],
@opencrvs/data-seeder:     extensions: { code: 'INTERNAL_SERVER_ERROR', exception: [Object] }
@opencrvs/data-seeder:   }
@opencrvs/data-seeder: ]
@opencrvs/data-seeder: error Command failed with exit code 1.
@opencrvs/data-seeder: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run seed exited 1 in '@opencrvs/data-seeder'
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```


I increased the tries to 10 

```
❯ yarn seed:dev
yarn run v1.22.4
$ ACTIVATE_USERS=true lerna run seed --stream --scope @opencrvs/data-seeder
lerna notice cli v5.6.2
lerna notice filter including "@opencrvs/data-seeder"
lerna info filter [ '@opencrvs/data-seeder' ]
lerna info Executing command in 1 package: "yarn run seed"
@opencrvs/data-seeder: $ ts-node -r tsconfig-paths/register src/index.ts
@opencrvs/data-seeder: Seeding roles
@opencrvs/data-seeder: Seeding locations
@opencrvs/data-seeder: Seeding users
@opencrvs/data-seeder: Trying again for time:  2
@opencrvs/data-seeder: Trying again for time:  3
@opencrvs/data-seeder: Trying again for time:  4
@opencrvs/data-seeder: Trying again for time:  5
@opencrvs/data-seeder: Seeding certificates
lerna success run Ran npm script 'seed' in 1 package in 18.5s:
lerna success - @opencrvs/data-seeder
✨  Done in 18.91s.
```

and it worked afterwards. Same if I increase the delay to 5 seconds

```
lerna notice cli v5.6.2
lerna notice filter including "@opencrvs/data-seeder"
lerna info filter [ '@opencrvs/data-seeder' ]
lerna info Executing command in 1 package: "yarn run seed"
@opencrvs/data-seeder: $ ts-node -r tsconfig-paths/register src/index.ts
@opencrvs/data-seeder: Seeding roles
@opencrvs/data-seeder: Seeding locations
@opencrvs/data-seeder: Seeding users
@opencrvs/data-seeder: Trying again for time:  2
@opencrvs/data-seeder: Trying again for time:  3
@opencrvs/data-seeder: Seeding certificates
lerna success run Ran npm script 'seed' in 1 package in 21.1s:
lerna success - @opencrvs/data-seeder
✨  Done in 21.44s.
```

(notice it barely managed to do it still)

**Review by rikukissa — APPROVED** [review]

**Pull request merged**